db/virtual_table.cc

```diff
@@ -61,4 +62,37 @@ bool virtual_table::contains_key(const dht::partition_range& pr, const dht::deco
     return pr.contains(dk, dht::ring_position_comparator(*_s));
 }
 
+mutation_source memtable_filling_virtual_table::as_mutation_source() {
+    return mutation_source([this] (schema_ptr s,
+        reader_permit permit,
+        const dht::partition_range& range,
+        const query::partition_slice& slice,
+        const io_priority_class& pc,
+        tracing::trace_state_ptr trace_state,
+        streamed_mutation::forwarding fwd,
+        mutation_reader::forwarding fwd_mr) {
+
+        if (fwd == streamed_mutation::forwarding::yes) {
+            throw std::runtime_error("streamed_mutation::forwarding::yes unsupported in virtual tables");
+        }
+
+        auto populate = [this, mt = make_lw_shared<memtable>(schema()), s, range, permit, slice, pc, trace_state, fwd, fwd_mr] () mutable {
+            return execute(*mt).then([this, mt, s, &range, &permit, &slice, &pc, &trace_state, &fwd, &fwd_mr] () {
```

Comment:
We should connect the memtable somehow to the `reader_permit` system, so that its size is accounted there.

Reply:
Maybe give execute() a proxy object which takes a mutation, and updates the permit after each apply().

Reply:
Wouldn't it be sufficient to update the permit with the summarized size of the memtable after execute finishes? The permit has anything to say only after execute anyway, right?

Reply:
`execute()` is a deferring operation. It can defer until it completes, and take arbitrary amount of time in general.

Reply:
So I think updating after execute() completes is too late.

Reply:
But since execution is inside `populate()`, and it is chained, wouldn't deferring it also defer the construction of the reader?

Reply:
It would, but why is it relevant? We already have a permit before we start execute(). We should account all memory used on behalf of the read on that permit.

Reply:
ah right, we receive the permit from the outside, my bad
inside execute it is then